FlyView: Update to 3.5 (#174)
mavlink/qgc-user-guide: en/FlyView/FlyView.md

# Fly View

The Fly View is the main view you will use while flying your vehicle. 

> You can switch between a map view and a video view (if available).

Author: @DonLakeFlyer Is this still true? I haven't got video test going.

![Fly View](../../assets/fly/fly_view_overview.jpg)

## Map

The map shows the positions of all connected vehicles and the  mission for the current vehicle.

You can drag the map to move it around (the map automatically recentres after a certain amount of time).

Once flying, you can click on the map to set a GoTo position.

## Fly Tools {#fly_tools}

On the left edge of the screen you will see the *Fly Tools*.
You can use these to takeoff, return, pause, or continue/start an action (e.g. a mission).

The order of tools from top to bottom is:

* Takeoff/Land
* Return To Launch
* Pause
* Action

## Video

> At the lower left of the display you will see video output.

Author: Is any of this section true, and if not, what is the story? I don't have video working.

Author: Yes it is true. Will update image.

*QGroundControl* supports RTP and RTSP video streaming over your vehicles UDP connection. 
It also support directly connected UVC device support. 
More details on QGC Video support can be found on the [Video README](https://github.com/mavlink/qgroundcontrol/blob/master/src/VideoStreaming/README.md).

By clicking on the video you can make it be the main display for the *Fly View*.

## Instrument Panel

To the right is an instrument panel showing you current information on your vehicle. 
By default this displays vehicle telemetry, but you can switch the content using the drop down menu.

The different content is listed below.

### Values (Telemetry)

The values page shows telemetry information.
The displayed information can be configured by clicking on the small gear icon.

![Instrument Page - for values/telemetry](../../assets/fly/instrument_page_values.jpg)

### Camera

The camera page can be used to manually trigger the camera.

![Instrument Page - for Camera](../../assets/fly/instrument_page_camera.jpg)

### Video Stream

> This page is used to enable and record video streaming.

Author: FYI this is based on what you see when video is not enabled. If you want something else, can you provide screenshots?

# Fly View

The Fly View is used to command and monitor the vehicle when flying.

> You can use it to:

Author: FYI, note high level description of what you can do, and forward links to those sections

You can continue the current mission while (unless already in a mission!):
1. Click the **Action** button on the *Fly Tools*
1. Select the *Continue Mission* action from the dialog.

   ![Continue Mission/Change Altitude action](../../assets/fly/continue_mission_change_altitude_action.jpg)

1. Drag the confirmation slider to continue the mission.

   ![Continue Mission](../../assets/fly/continue_mission.jpg)

#### Resume Mission {#resume_mission}

*Resume Mission* is used to resume a mission after performing an [RTL/Return](#rtl) or [Land](#land) from within a mission (in order, for example, to perform a battery change).

> **Note** If you are performing a battery change, **do not** disconnect QGC from the vehicle after disconnecting the battery.
After you insert the new battery *QGroundControl* will detect the vehicle again and automatically restore the connection. 

After reconnecting you will be prompted with a *Resume Mission* confirmation slider.
If you want to resume the mission, confirm this and the mission will start from the last waypoint traveled through. 

> Once the mission is rebuilt you will be presented with another Resume Mission slide which allows you to review the rebuilt mission before starting it again.

Author: I presume that this appears after I press the button to resume from waypoint in the diagram below? Does the mission actually appear to change? (ie do you really get to review changes?)

Confirm this *Resume Mission* slider to continue on with the mission.

> ![Resume Mission](../../assets/fly/resume_mission.jpg)

Author: Note, the image here is from a Land. Couldn't get RTL to pop up option - it just gave me option to remove the plan. This is on SITL and jmavsim.

Reviewer: > Couldn't get RTL to pop up option

Not quite sure what you mean? Are you talking about not getting a resume mission to show up? You need to RTL from the middle of a mission. You have to be at least a couple waypoints away from the end when you do it.

Author: Yes. I thought I was - but in any case it all works now. Thanks.

You can configure what information is display by clicking the small gear icon on the top left of the panel.
Each value can be displayed in normal or "large" size (large size shows just one value per row in the page, while normal shows 2).

![Instrument Page - values settings](../../assets/fly/instrument_page_values_settings.jpg)

### Camera

The camera page is used to configure and control the camera.
For a camera connected directly to the Flight Controller the only available option is camera triggering:

![Instrument Page - for Camera](../../assets/fly/instrument_page_camera.jpg)

When connected to camera that supports the [MAVLink Camera Protocol](https://mavlink.io/en/services/camera.html) you can additionally configure and use other camera services that it makes available.
For example, if your camera supports video mode you will be able to  switch between still image capture and video mode, and start/stop recording.

![Instrument Page - Camera MAVLink Settings](../../assets/fly/instrument_page_camera_mavlink.jpg)

Advanced settings can be changed by clicking the gear icon at the top left of the page.

> The settings that are displayed will depend on/are a property of the camera.

Author: @dogmaphobic In FlyView, if you're using the MAVLink camera you get option to configure settings. Are these dynamic based on the config file, or hard coded? (I have said here dynamic)

Reviewer 2: Most settings come dynamically from the camera. That's what comes in the _camera definition file_ (the XML file). There are a few hardcoded, common settings at the bottom:

- Photo Mode (Single/Time Lapse)
- Photo Interval (if Time Lapse)
- Reset Camera Defaults (sends a reset command to the camera)
- Format (storage)

Author: Thanks!

- **Map:** Displays the positions of all connected vehicles and the mission for the current vehicle.
  - You can drag the map to move it around (the map automatically re-centres after a certain amount of time).
  - Once flying, you can click on the map to set a [Go to](#goto) or [Orbit at](#orbit) location.
- **Fly Toolbar:** Key status information for sensors (GPS, battery, RC control), and vehicle state (Flight mode, Armed/Disarmed status).
  - Select the sensor indicators to view more detail.
  - Click the *Flight mode* text (e.g. "Hold") to select a new mode. Not every mode may be available.
  - Click the *Armed/Disarmed* text to toggle the armed state. If flying you can click this to Emergency Stop.
- **Fly tools:** You can use these to:
  - Toggle between takeoff/land.
  - Pause/restart the current operation (e.g. landing, or the mission).
  - Safety return (also known as RTL or Return).
  - The *Action* button offers other appropriate options for the current state (these overlay the *Confirmation Slider*). 
    Actions include changing the altitude or continuing a mission.
  - Enable the [preflight checklist](#preflight_checklist) (tool option disabled by default).

> - **[Instrument Panel](#instrument_pane):** A multi-page widget that displays vehicle information including: telemetry, camera, video, system health, and vibration.

Reviewer 3: ```suggestion
- **[Instrument Panel](#instrument_panel):** A multi-page widget that displays vehicle information including: telemetry, camera, video, system health, and vibration.
```

- **[Video/Switcher](#video_switcher):** Toggle between video or map in a window.
  - Click the element to switch *Video* and *Map* to foreground.
  - *QGroundControl* supports RTP and RTSP video streaming over your vehicles UDP connection. 
    It also support directly connected UVC device support.
    QGC video support is further discussed in the [Video README](https://github.com/mavlink/qgroundcontrol/blob/master/src/VideoStreaming/README.md).
- **Confirmation Slider:** Context sensitive slider to confirm requested actions. 

>   Slide to start operation. Press **X** to cancel.

Reviewer 4: ```suggestion
  Slide to start operation. Click**X** to cancel.
```

Reviewer 3: It's not a click if it's on mobile is it?

Reviewer 4: "Press" actually made me think of pressing the "X" key in my keyboard for a few seconds.
"Press the **X** Icon to cancel" would be fine too.
But this page currently has 24 mentions of "click" and only this one of "press".

Author: The whole library uses click because I test and write it on a desktop. Press is more generic, so I will change this document, and try to change other cases as I go along.

There are a number of other elements that are not displayed by default/are only displayed in certain conditions. 
For example, the multi-vehicle selector is only displayed if you have multiple vehicles, and the preflight checklist tool button is only displayed if the appropriate setting is enabled.

> ## Instrument Panel {#instrument_pane}

Reviewer 3: ```suggestion
## Instrument Panel {#instrument_panel}
```

To takeoff (when landed):
1. Click the **Takeoff** button in the *Fly Tools* (this will toggle to a **Land** button after taking off).
1. Optionally set the takeoff altitude in the right-side vertical slider.
1. Confirm takeoff using the slider.

![takeoff](../../assets/fly/takeoff.jpg)

### Land {#land}

You can land at the current position at any time while flying:
1. Click the **Land** button in the *Fly Tools* (this will toggle to a **Land** button when landed).
1. Confirm landing using the slider.

![land](../../assets/fly/land.jpg)

### RTL/Return

Return to the home position at any time while flying:
1. Click the **RTL** button in the *Fly Tools*.
1. Confirm RTL using the slider.

![land](../../assets/fly/land.jpg)

> **Note** The vehicle may also land at the home position, depending on its type and configuration.

### Change Altitude {#change_altitude}

You can change altitude while flying, except when in a mission:
1. Click the **Action** button on the *Fly Tools*
1. Select the *Change Altitude* action from the dialog.

   ![Continue Mission/Change Altitude action](../../assets/fly/continue_mission_change_altitude_action.jpg)

> 1. Move the horizontal slider to the desired altitude, then drag the confirmation slider to start the action.

Reviewer 4: ```suggestion
1. Move the vertical slider to the desired altitude, then drag the confirmation slider to start the action.
```

* [MAV_CMD_DO_SET_CAM_TRIGG_DIST](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_CAM_TRIGG_DIST)
* [MAV_CMD_DO_FENCE_ENABLE](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_FENCE_ENABLE)
* [MAV_CMD_IMAGE_START_CAPTURE](https://mavlink.io/en/messages/common.html#MAV_CMD_IMAGE_START_CAPTURE)
* [MAV_CMD_IMAGE_STOP_CAPTURE](https://mavlink.io/en/messages/common.html#MAV_CMD_IMAGE_STOP_CAPTURE)
* [MAV_CMD_VIDEO_START_CAPTURE](https://mavlink.io/en/messages/common.html#MAV_CMD_VIDEO_START_CAPTURE)
* [MAV_CMD_VIDEO_STOP_CAPTURE](https://mavlink.io/en/messages/common.html#MAV_CMD_VIDEO_STOP_CAPTURE)
* [MAV_CMD_DO_CHANGE_SPEED](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_CHANGE_SPEED)
* [MAV_CMD_NAV_TAKEOFF](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_TAKEOFF)

#### Remove Mission Prompt After Landing {#resume_mission_prompt}

You will be prompted to remove the mission from the vehicle after the mission completes and the vehicle lands and disarms. 
This is meant to prevent issues where stale missions are unknowingly left on a vehicle, potentially resulting in unexpected behavior.

> ### Display Video {#video_switcher}

Author: @Williangalvani I added two sections here for displaying and recording video. Can you please review. The camera and video streaming instrument pages above were also updated.

> *QGroundControl* supports RTP and RTSP video streaming over your vehicle's UDP connection.

Reviewer 3: QGC supports several video stream options:

RTP/RTSP
Bare UDP
Taisync
Local video devices (like a usb fpv video receiver)

Reviewer 2: Add TCP-MPEG2 and MPEG-TS to the list.
Also note that UVC is only supported on desktop platforms (no Android or iOS)

Author: @dogmaphobic @jaxxzer Thanks. I have replaced this with the following. Can you sanity check (ie I don't know anything about most of these, so not sure if they are correctly described in context. This is also why I omitted bare UDP - that doesn't make sense to me as a video stream option. 

----

*QGroundControl* supports a number of video streaming options:
- RTP and RTSP video streaming over your vehicle's UDP connection
- Taisync
- TCP-MPEG2 and MPEG-TS
- USB video devices (UVC) on desktop platforms (Android or iOS not supported).
  This includes webcams and USB FPV video receivers.

Author: Actually, please ignore that. This is not the right section to talk about what streaming we support. If needed we should update https://docs.qgroundcontrol.com/en/SettingsView/General.html#video

When video streaming is enabled the stream is displayed in the video switcher window at the bottom left of the map.
Click the switcher to toggle *Video* and *Map* to foreground.

![Video Stream Record](../../assets/fly/video_record.jpg)

> > **Note** The vidoe stream is configured/enabled in [Application Settings > General tab > Video](../SettingsView/General.md#video).

Reviewer 3: ```suggestion
> **Note** The video stream is configured/enabled in [Application Settings > General tab > Video](../SettingsView/General.md#video).
```

> It is possible to resize the video switcher.

Author: @Williangalvani As far as I can see you can resize and hide the switcher, but you can't move it - is that correct?

Reviewer 4: Not in there, only if you detach it:
![Peek 2019-03-18 10-54](https://user-images.githubusercontent.com/4013804/54535324-12f7bf80-496d-11e9-8d80-cde156cff9eb.gif)

Author: Awesome thanks! I have updated the section appropriately.

First click on its top-left corner.
Resize the window by dragging the icon in the top right corner, and hide it with the icon in the lower left.

![Video Pop](../../assets/fly/video_pop.jpg)

### Record Video

> If supported by the camera, *QGroundControl* can start and stop video recording on the camera itself, or it can record and save the video stream locally.

Reviewer 3: ```suggestion
If supported by the camera and vehicle, *QGroundControl* can start and stop video recording on the camera itself. *QGroundControl* can also record the video stream and save it locally.
```

> **Tip** Video stored on the camera may be of much higher quality, but it is likely that your ground station will have a much larger recording capacity.

#### Record Video Stream (on GCS)

> Record the video stream on the [video stream instrument page](#video_instrument_page).

Reviewer 3: ```suggestion
Video stream recording is controlled on the [video stream instrument page](#video_instrument_page).
```

> Click the red circle to start recording a new video (a new video is started every time the circle is clicked); the circle will change into a red square while recording is in progress.

Reviewer 3: ```suggestion
Click the red circle to start recording a new video (a new video file is created each time the circle is clicked); the circle will change into a red square while recording is in progress.
```
